### ISSUE TITLE
[ON HOLD #30561] [$500]  Attachment - Hmm its not here page displayed on login with receipt attachment URL

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#28568]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Click on plus and raise manual request money request with any amount and any user
3. Open the request money page by clicking twice on IOU message
4. edit URL and add '/edit/receipt' in it
5. Upload any image and after upload, open it
6. Copy the URL
7. Logout and visit the copied URL
8. Login and observe that it displays 'Hmm its not here' page

## Expected Result:
 App should display receipt image on login with receipt attachment

## Actual Result:
App displays 'Hmm its not here' on login with receipt attachment

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] Windows / Chrome

**Version Number:** 1.3.75-8

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/20116bef-5bd5-49cb-9d6f-ed7394100995

https://github.com/Expensify/App/assets/78819774/f5d00b21-598f-4c86-b539-9d2d671fc480

https://github.com/Expensify/App/assets/78819774/7c45410f-0f38-453a-b9a3-fb7bab2b329e

https://github.com/Expensify/App/assets/78819774/825cbe0c-3550-42b9-81fc-45f8f026b5bc

https://github.com/Expensify/App/assets/78819774/f745d5fa-ea43-4fd0-abb5-3d5e4132229f

**Expensify/Expensify Issue URL:**

**Issue reported by:** @dhanashree-sawant

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1696066001795669

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019998a84689b09070</li>
        <li>Upwork Job ID: 1708585513587392512</li>
        <li>Last Price Increase: 2023-10-22</li>
    </ul>
</details>

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The attachment is not shown after logging in with the attachment URL.
### What is the root cause of that problem?
The report and the report action elements are empty in the beginning causing the error to show up
https://github.com/Expensify/App/blob/4ebeea00ebf1470bbf085a51cb9b9ec0735cf511/src/pages/home/report/ReportAttachments.js#L30-L31
### What changes do you think we should make in order to solve the problem?

we should use `useOnyx` to dynamically fetch the value of the report and the reportActions in the `ReportAttachments` component and pass it to the `AttachmentModal` page:

change the `ReportAttachments` to fetch the report and reportactions from `useOynx`

```javascript
const propTypes = {
    /** Navigation route context info provided by react navigation */
    route: PropTypes.shape({
        /** Route specific parameters used on this screen */
        params: PropTypes.shape({
            /** The report ID which the attachment is associated with */
            reportID: PropTypes.string.isRequired,
            /** The uri encoded source of the attachment */
            source: PropTypes.string.isRequired,
        }).isRequired,
    }).isRequired,
    report: reportPropTypes,
    reportActions: PropTypes.shape(reportActionPropTypes),
};

const defaultProps = {
    report: {},
    reportActions: {},
};

function ReportAttachments(props) {
    const reportID = _.get(props, ['route', 'params', 'reportID']);
    const source = decodeURI(_.get(props, ['route', 'params', 'source']));
    return (
        <AttachmentModal
            allowDownload
            defaultOpen
            report={props.report}
            reportActions={props.reportActions}
            source={source}
            onModalHide={() => Navigation.dismissModal(reportID)}
            onCarouselAttachmentChange={(attachment) => {
                const route = ROUTES.REPORT_ATTACHMENTS.getRoute(reportID, attachment.source);
                Navigation.navigate(route);
            }}
        />
    );
}

ReportAttachments.propTypes = propTypes;
ReportAttachments.defaultProps = defaultProps;
ReportAttachments.displayName = 'ReportAttachments';

export default withOnyx({
    report: {
        key: ({route}) => `${ONYXKEYS.COLLECTION.REPORT}${route.params.reportID}`,
    },
    reportActions: {
        key: ({route}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${route.params.reportID}`,
        canEvict: false,
    },
})(ReportAttachments);
```

where in the `AttachementModel` we should add the `reportActions` to the `propTypes`:
```javascript
    reportActions: PropTypes.shape(reportActionPropTypes),
```
and also in the `defaultProps`:

```javascript
    reportActions: {},
```


Now we need to update the `AttachmentCarousel` condition to show only if the report and the report actions are not empty from :
https://github.com/Expensify/App/blob/fbd04cf6afe7e7de9a666a5450611253aa4b7a5a/src/components/AttachmentModal.js#L381-L389

To:
```javascript
       {!_.isEmpty(props.report) && !_.isEmpty(props.reportActions) ? (
                        <AttachmentCarousel
                            report={props.report}
                            onNavigate={onNavigate}
                            reportActions={props.reportActions}
                            source={props.source}
                            onClose={closeModal}
                            onToggleKeyboard={updateConfirmButtonVisibility}
                            setDownloadButtonVisibility={setDownloadButtonVisibility}
                        />
```


### POC

**Comment by CortneyOfstad** [comment]
@sobitneupane thoughts on the proposals above?

**Comment by CortneyOfstad** [comment]
@sobitneupane bump ^^^

**Comment by sobitneupane** [comment]
Sorry for the delay.

I will try to review the proposals before EOD tomorrow

**Comment by sobitneupane** [comment]
@dukenv0307 Thanks for your proposal.

> Since the ReportAttachments and AttachmentModal don't need the full report object (only the reportID is needed), we should modify them to pass through only the reportID rather than full report object.

We are using `report.parentReportActionID` in AttachmentModal.

> This will make sure it renders correctly if we go to offline mode right after logging in.

How are we handling attachments in offline mode as of now if they are not already loaded?

**Comment by sobitneupane** [comment]
> Yeah it's a new code update, we'll need to connect to the report as well in AttachmentModal then.

@dukenv0307 Can you please update your proposal to reflect the update? 

> currently it will show loading indicator when offline if the attachment is not yet ready. When we go online the attachment will show.

In that case, we can have the same behavior here as well.

**Comment by CortneyOfstad** [comment]
@sobitneupane bump on @dukenv0307's comment above ^^^ — TIA!

**Comment by CortneyOfstad** [comment]
I am heading OoO until Oct. 25, so reassigning the BZ label to keep an eye on things while I am gone 👍

**Comment by lschurr** [comment]
What's the latest on this one @sobitneupane?

**Comment by lschurr** [comment]
Great! @situchan I'll assign you. Could you confirm the latest on this issue?

**Comment by situchan** [comment]
I am not able to reach out this issue because deeplink (/r/:reportID/edit/receipt) doesn't work at all. Seems like recent regression.

<img width="603" alt="Screenshot 2023-10-29 at 7 08 07 PM" src="https://github.com/Expensify/App/assets/108292595/c5b3ba90-656c-496f-9110-c27bf71729b5">


https://github.com/Expensify/App/assets/108292595/52df36c8-8708-4398-961f-38b8432d592b

**Comment by situchan** [comment]
@dukenv0307's [proposal](https://github.com/Expensify/App/issues/28568#issuecomment-1742363882) looks good to me.
🎀 👀 🎀 C+ reviewed

> * Since the `ReportAttachments`  don't need the full `report` object (only the reportID is needed), we should modify it to pass through only the `reportID` to `AttachmentModal` rather than full report object. This will minimize component rerendering due to report/reportActions change
> * In `AttachmentModal`, connect to `report` via Onyx (from the reportID passed as prop)
> * In `AttachmentCarousel`, connect to `report` via Onyx (from the reportID passed as prop)

@dukenv0307 above changes are redundant and don't help rendering optimization. Instead, connect to `report` via Onyx in `ReportAttachments`.

**Comment by CortneyOfstad** [comment]
@situchan update on the PR review above? Thanks!

**Comment by CortneyOfstad** [comment]
@situchan bump ^^^ thanks!

**Comment by situchan** [comment]
PR is in review

**Comment by mountiny** [comment]
PR not on hold anymore @dukenv0307 @situchan lets get this over the finish line, thanks!

**Comment by situchan** [comment]
@mountiny please check https://github.com/Expensify/App/pull/30561#issuecomment-1849897340. Should we prioritize this over https://github.com/Expensify/App/issues/31105?

**Comment by mountiny** [comment]
Thanks, I am bumping the PR this is on hold for

**Comment by CortneyOfstad** [comment]
Any update @mountiny? Thanks!

**Comment by mountiny** [comment]
PR is still on hold

**Comment by mountiny** [comment]
@situchan @dukenv0307 is this still on hold?

**Comment by CortneyOfstad** [comment]
I just realized the title was never updated for the on-hold PR (https://github.com/Expensify/App/pull/30561#issuecomment-1849897340), so updated to better reflect that. 

I did a bit of a deep dive and found that the issue was on hold because of another PR (https://github.com/Expensify/App/pull/28999) which was merged about 3 weeks ago. @situchan @dukenv0307 I bumped you both on your PR. Let me know where we are at with that 👍

**Comment by situchan** [comment]
Updated on PR

**Comment by CortneyOfstad** [comment]
@situchan checking the PR, it shows that @dukenv0307 mentioned an issue last week affecting main. Do either of you have an update on the status of the PR? Do you think that this is something that can be resolved by EOW? Thanks!

**Comment by situchan** [comment]
Seems like we're currently blocked on unrelated bug

**Comment by CortneyOfstad** [comment]
@situchan bumped you on the [on-hold issue](https://github.com/Expensify/App/pull/30561#issuecomment-1849897340), as it looks like the review is waiting for you. Can you please provide an update there by EOD?

**Comment by CortneyOfstad** [comment]
PR has some movement in the last few hours, so looking good there! Will continue to keep an eye on it 👍

**Comment by mountiny** [comment]
@dukenv0307 @situchan what should we do here, the issue is on hold for the PR which you can now close @dukenv0307

**Comment by CortneyOfstad** [comment]
Discussed with @mountiny and due to the nature of the [original PR](https://github.com/Expensify/App/pull/30561#issuecomment-1849897340) we decided a partial payment of 75% is appropriate here. Normally, no payment would be made, as another PR was utilized. However, due to the time and effort put into the original PR, we both felt that payment was appropriate. 


@situchan @dukenv0307 — I sent you both offers in Upwork for the adjusted contract. Please let me know once you accept and i can get those paid ASAP today. Thank you!

**Comment by CortneyOfstad** [comment]
@dukenv0307 It is showing that it is still waiting for you to accept the offer in Upwork — can you let me know once you accept and I can get that paid to you ASAP. Thanks!

**Comment by dhanashree-sawant** [comment]
Hello @CortneyOfstad @mountiny, I tried but couldn't understand whether the issue was worked on somewhere or not so can you let me know if it is eligible for reporting bonus or not?

**Comment by CortneyOfstad** [comment]
@dhanashree-sawant Sorry about that! I've sent you a request in Upwork — please let me know once you accept and I will get that paid ASAP. Thanks!

**Comment by CortneyOfstad** [comment]
## Payment Summary

@dhanashree-sawant (reporter) – paid $50 via Upwork
@dukenv0307 (contributor) — paid $375 via Upwork
@situchan (C+) — paid $375 via Upwork
